blt/middleware/ip_restrict.py

```diff
@@ -0,0 +1,35 @@
+from django.core.cache import cache
+from django.http import HttpResponseForbidden
+
+from website.models import BlockedIP
+
+
+class IPRestrictMiddleware:
+    """
+    Middleware to restrict access based on client IP addresses.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def blocked_ips(self):
+        blocked_ips = cache.get("blocked_ips")
+        if blocked_ips is None:
+            # Retrieve all blocked IP addresses from the database
+            blocked_addresses = BlockedIP.objects.values_list("address", flat=True)
+            blocked_ips = set(blocked_addresses)
+            cache.set("blocked_ips", blocked_ips, timeout=86400)
+        return blocked_ips
+
+    def __call__(self, request):
```

Comment:
Isnt blocking the user agent itself is ineffective as it will block it for every user with that user agent and the spammer can always use other device , why would we need it ?